### ISSUE TITLE
bug: Unable to use any 'this' property inside isDateEnabled function

### ISSUE THREAD

**Issue by Rodrigocambraia14** [ionic-team/ionic-framework#25189]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x
- [ ] Nightly

### Current Behavior

When trying to use any 'this' property inside IsDateEnabled function, it aways returns undefined. 

I already tried make the component wait till the required properties that I wanted were previously loaded, but when the function is called by the component, all the global variables just doesn't exist. 

### Expected Behavior

Using the globalVariables defined inside functions, like any other function you would call inside your .ts file.

If I made some huge mistake, I'm previously sorry, and also asking for help.

### Steps to Reproduce

Even if you block the Ion-dateTime component from existing with *ngIf until the wanted property receives the desired value (an array of dates from an API ), there will be no properties available to use inside the function below:

isDateEnabled(dateIsoString: string) {
    const date = new Date(dateIsoString);

    if (this == undefined || this.bookedDates == undefined || this.bookedDates.length == 0)
      return true;
    
    return this.bookedDates.some(x =>this.formatDate(x.startDate) == date );
  }

### Ionic Info

   Ionic CLI                     : 6.17.1 (C:\Users\fesco\AppData\Roaming\npm\node_modules\@ionic\cli)
   Ionic Framework               : @ionic/angular 6.1.2
   @angular-devkit/build-angular : 13.3.0
   @angular-devkit/schematics    : 11.2.18
   @angular/cli                  : 13.3.0
   @ionic/angular-toolkit        : 3.1.1

### Additional Information

I tried almost all the available  examples from the last PullRequest Implemented about ion-Datetime

**Comment by sean-perkins** [comment]
Thanks for the issue @Rodrigocambraia14.

You have two options available to you.

If you want to use the function declaration pattern:
```ts
constructor() {
  this.isDateEnabled = this.isDateEnabled.bind(this);
}

isDateEnabled(dateString: string) {
  console.log('this', this);
  return true;
}
```

or you can open the scope, so that it passes through:

```ts
isDateEnabled = (dateString: string) => {
  console.log('this', this);
  return false;
}
```

I've highlighted both examples here: https://stackblitz.com/edit/angular-twk2ev?file=src%2Fapp%2Fapp.component.ts

Can you let me know if this resolves your issue?

**Comment by Rodrigocambraia14** [comment]
I tried the first example and it worked perfectly !! Thank you so much !

**Comment by Rodrigocambraia14** [comment]
The only problem that I'm still having, is that for some unkown reason the calendar is blocking one day after the day that sould be blocked. I already debbuged but it seems to be flowing in a correctly logical way, as it returns false for the days that are supposed to be blocked, and true to the others. 

I linked an image that show the logs and the calendar behaviour above.

If its possible to help in any way, I'm already greatfully thankful.
![image](https://user-images.githubusercontent.com/77635828/165187080-f2879ad4-af9d-4433-b85f-bc4ae90cdc24.png)
